### ISSUE TITLE
read_csv does not fail gracefully if tempdir is filled/full when reading gzipped files. 

### ISSUE THREAD

**Issue by JustGitting** [tidyverse/readr#1055]
R version 3.5.3 (2019-03-11)
Platform: x86_64-pc-linux-gnu (64-bit)
readr: 1.3.1.9000

I compiled readr from source (hence at version 1.3.1.9)  to see if the problem went away, but it did not make any difference. 

Hi,

This problem stumped me for a few days as I thought I had hardware and or system library problems.

I was using readr::read_csv() to open a large gzipped CSV file with about 500,000 rows and 300 columns. I regularly check the size of the returned data.frame as I've been performing multi-stage processing/cleaning and save the data between stages.

For example:

```r
df <- readr::read_csv('mydata.csv.gz', guess=10000)
```

On one occasion, I found that the data frame was only about 200,000 rows and I was getting one or two warnings about parsing the file. 

I thought I might have accidentally removed too many rows in a previous stage or used the wrong delimiter somehow. I unzipped the csv.gz file from a bash terminal and read the file in using base R read_csv(). Much to my relief I found that all the expected number of rows and columns were present, so the csv.gz file was fine.

The odd thing was that almost every time I ran the readr::read_csv() function to read the csv.gz file, the row and column that produced the parsing error was different. It was never consistent. Hence, I thought I had some sort of hardware problem or an odd system library issue.

I eventually found that my /tmp directory was almost full. Such that readr:read_csv() would fill the /tmp folder and I would get the parsing errors and a smaller tibble than expected.

Below is a reproducible example for a GNU/Linux system. 

We can create a small temporary directory using tmpfs as root to illustrate the problem.

```shell
# mkdir /tmp_r
# mount -t tmpfs -o size=1M tmpfs /tmp_r/
# df
Filesystem           Type      Size  Used Avail Use% Mounted on
...
tmpfs                tmpfs     1.0M     0  1.0M   0% /tmp_r

```

From my understanding, readr decompresses files to the tempdir() location so it can parse the file and guess the column types. To manually override the tempdir() location, I found the [unixtools](https://github.com/s-u/unixtools) library to be the easiest approach. We'll also change the temporary session path (R_SESSION_TMPDIR) for good measure.

```r
library(devtools)
devtools::install_github("s-u/unixtools")
library(unixtools)
unixtools::set.tempdir('/tmp_r')
tempdir()
# [1] "/tmp_r"
Sys.setenv(R_SESSION_TMPDIR = '/tmp_r')
Sys.getenv('R_SESSION_TMPDIR')
# [1] "/tmp_r"
```

Create a sufficiently large data.frame and save it as a CSV file.

```r
df <- data.frame(matrix(0, nrow=1000000, ncol=20))
write.csv(df, 'mytable.csv', row.names = F, quote = T )

```

Read the csv file back into R to check if it worked 

```r
df_csv <- read.csv( 'mytable.csv' )
dim(df_csv)
# [1] 1000000      20
```

I could have used write.csv(gzfile()) to compress the file, but it's easier do it manually in the terminal.

```shell
$ gzip -c mytable.csv > mytable.csv.gz
```

Base read.csv() and gzfile() has no problem reading the file. This is important because readr also uses gzfile() and I wanted to rule out gzfile() being the cause of the problem.

```r
df_csv_gz <- read.csv( gzfile('mytable.csv.gz') )
dim(df_csv_gz)
# [1] 1000000      20
```

However, running readr:read_csv() produces warnings and a tibble with fewer rows than expected.
```r
df_csv_gz <- readr::read_csv( 'mytable.csv.gz' )
# Parsed with column specification:
#   cols(
#     .default = col_double()
#   )
# See spec(...) for full column specifications.
# Warning: 1 parsing failure.
# row col   expected     actual         file
# 26212  -- 20 columns 13 columns <connection>
dim(df_csv_gz)
# [1] 26212    20

```
The tempdir() is not completely filled on the first run as you would expect.
If you re-running the above command two times tempdir is filled with three files.

```shell
# ls -l /tmp_r
total 1.0M
-rw-r--r-- 1 user1 user1 1016K Dec 26 12:12 file45439fd235a
-rw-r--r-- 1 user1 user1  8.0K Dec 26 12:12 file4544bae6c09
-rw-r--r-- 1 user1 user1     0 Dec 26 12:12 file454604032cd
```

Running the read_csv() for a forth time results in an empty data.frame and no messages at all.
```r
df_csv_gz <- readr::read_csv( 'mytable.csv.gz' )
dim(df_csv_gz)
# [1] 0 0
```

If we increase the \tmp_r directory size, the problem is solved.

```shell
# rm -rf /tmp_r/*
# mount -t tmpfs -o remount,size=50M tmpfs /tmp_r/
# df
Filesystem           Type      Size  Used Avail Use% Mounted on
...
tmpfs                tmpfs     50M     0  50M   0% /tmp_r
```

```r
df_csv_gz <- readr::read_csv( 'mytable.csv.gz' )
# Parsed with column specification:
#   cols(
#     .default = col_double()
#   )
# See spec(...) for full column specifications.
dim(df_csv_gz)
# [1] 1000000      20

```
I would have expected readr:readr() to check if gzfile() successfully decompressed the file before trying to parse it. 

If you need any further information, please let me know.

**Comment by muschellij2** [comment]
On a similar note, would it be possible to have a `cleanup` option to delete the unzipped file in `tempdir` after reading to keep size of `tempdir` down?

**Comment by jimhester** [comment]
Fixed by #1172, vroom allows you to fully control where the temporary files are stored.
